site/pages/view-all-disruptions.page.tsx

```diff
@@ -146,34 +551,51 @@ const ViewAllDisruptions = ({ disruptions, newDisruptionId }: ViewAllDisruptions
 };
 
 export const getServerSideProps = async (): Promise<{ props: ViewAllDisruptionsProps }> => {
+    const services: Service[] = await fetchServices({ adminAreaCode: ADMIN_AREA_CODE });
+
     const data = await getPublishedDisruptionsDataFromDynamo();
 
     if (data) {
         const sortedDisruptions = sortDisruptionsByStartDate(data);
         const shortenedData: TableDisruption[] = sortedDisruptions.map((disruption) => {
             const modes: string[] = [];
-            const severities: Severity[] = [];
+            const severitys: Severity[] = [];
+            const serviceIds: string[] = [];
+            const operators: Operator[] = [];
 
             if (disruption.consequences) {
                 disruption.consequences.forEach((consequence) => {
-                    severities.push(consequence.disruptionSeverity);
-                    modes.push(splitCamelCaseToString(consequence.vehicleMode));
+                    const modeToAdd = getDisplayByValue(VEHICLE_MODES, consequence.vehicleMode) as string;
+                    if (!modes.includes(modeToAdd)) {
```

Comment:
If we check for the existence of modeToAdd we wouldn't need to cast to a string on the line above:
```ts
const modeToAdd = getDisplayByValue(VEHICLE_MODES, consequence.vehicleMode);
if (modeToAdd && !modes.includes(modeToAdd)) {
```